src/main/java/mendel/discretetask/Event.java

```diff
@@ -0,0 +1,162 @@
+     *
+     * @param rawDescription The raw description string containing the event description, start date, and end date.
+     * @return A new Event object with the parsed description, start date, and end date.
+     */
+    public static Event of(String rawDescription) {
+        String[] description = parseDescription(rawDescription);
+        return new Event(description[0], description[1], description[2]);
+    }
+
+    /**
+     * Factory method to load a Event object from stored data.
+     *
+     * @param mark A boolean indicating whether the database task was marked as done.
+     * @param description The database description of the event.
+     * @param from The database start date of the event in a formatted string.
+     * @param to The database end date of the event in a formatted string.
+     * @return An Event object initialized with the provided parameters.
+     */
+    public static Event loadOf(boolean mark, String description, String from, String to) {
+        Event initObj = new Event(description, from, to);
+        if (mark) {
+            initObj.markAsDone();
+        } else {
+            initObj.markAsUnDone();
+        }
+        return initObj;
+    }
+
+    /**
+     * Parses the raw description string to extract the event description, start date, and end date.
+     *
+     * @param rawDescription The raw description string containing the event description, start date, and end date.
+     * @return An array where the first element is the event description, the second element is the start date,
+     *         and the third element is the end date.
+     * @throws MendelException if there are issues with the format or content of the raw description.
+     */
+    private static String[] parseDescription(String rawDescription) {
+        handleError(rawDescription);
+        String[] slashSegments = rawDescription.split(" /from ");
+        String[] mainMessage = slashSegments[0].split(" ");
+        String startMsg = new DateTimeManager(slashSegments[1].split(" /to ")[0]).toString();
+        String endMsg = new DateTimeManager(slashSegments[1].split(" /to ")[1]).toString();
+        String reformattedMsg = "";
+        for (int i = 1; i < mainMessage.length; i++) {
+            if (i == mainMessage.length - 1) {
+                reformattedMsg += mainMessage[i];
+            } else {
+                reformattedMsg += mainMessage[i] + " ";
+            }
+        }
+        return new String[]{reformattedMsg, startMsg, endMsg};
+    }
+
+    /**
+     * Validates the raw description string and handles various error conditions.
+     *
+     * @param rawDescription The raw description string to be validated.
+     * @throws MendelException if there are issues such as missing description,
+     *                         missing start or end date, or incorrect formatting.
+     */
+    private static void handleError(String rawDescription) throws MendelException {
```

Comment:
Is it possible to change this exception to handleException to adhere to the coding standard?